Fixed altoclef crashing if quitting the world while the task is running.

MarvionKirito opened the pull request:
## Summary

<!-- What is this pull request for? Does it fix any issues? -->
Fixed altoclef crashing if quitting the world while the task is running.
## Information

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, examples, ...).

## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.

MarvionKirito committed:
1.19.2 Update

MarvionKirito committed:
Remove cave and void air in blocksToAvoid.

MarvionKirito committed:
Fixes and Additions:
0. The bot now doesn’t use a block needed for resources.
1. The bot now has a settings dontThrowAwayEnchantedItems.
2. Added Totem of Undying in the default settings of importantItems.
3. Added Hero task. See @help to see the command and description.

JustaSqu1d committed:
chore: document "hero" in usage.md

MarvionKirito committed:
Fixes:
0. Fix altoclef crashing when entering the @get command.
1. Fix Hero task doesn't do anything.

MarvionKirito committed:
Merge pull request #1 from JustaSqu1d/patch-2

chore: document "hero" in usage.md

MarvionKirito committed:
Fix:
0. Fix Hero task getting stuck sometimes.

MarvionKirito committed:
Fix:
0. Water and lava block detection.

Optimizations:
0. In getting a block, the bot checks if the block is present before dimension checking.
1. Optimized block tracking.
2. In mining a block, the bot now uses a tool that is fastest to mine a block.

MarvionKirito committed:
Fix:
0. Fixed after beating the game with the marvion version. It doesn't finish sometimes.

Additions:
0. The bot now gets XP and drops in the hero task.
1. @get sapling is now available. You can also specify like @get oak_sapling.

Optimization:
0. Optimized hero task.

MarvionKirito committed:
Fix:
0. Added warden and wither to the dangerous mob.

Optimization:
0. Optimized food chain.

MarvionKirito committed:
Optimizations:
0. Optimized do to entity task.
1. Optimized kill dragon with bed task.
2. Optimized render distance manipulation.
3. Optimized mob defense chain.
4. Optimized locate stronghold task.

MarvionKirito committed:
Optimization:
0. Optimized construct nether portal tasks.

MarvionKirito committed:
Fix:
 - Fixed compatibility between altoclef and meteor client.

MarvionKirito committed:
Fix:
 - Fixed compatibility between altoclef and meteor client.

MarvionKirito committed:
1. Updated to 1.19.3

JustaSqu1d committed:
perf: equip weapon then check charge progress

This saves some durability and time.

MarvionKirito committed:
Merge pull request #4 from JustaSqu1d/patch-1

perf: equip weapon then check charge progress

JustaSqu1d committed:
chore(README): update versions availiblity

MarvionKirito committed:
Additions:
1. @ coverwithsand (Cover lava with sands in nether).
2. @ coverwithblocks (Cover lava with blocks in nether).
3. @ meat (Get food that consists only of meat, without any vegetables).

Fixes:
1. ConcurrentModificationException
2. Fixed compatibility between meteor client.
3. Some fixes.

Improvements:
1. Some improvements.

MarvionKirito committed:
Merge pull request #5 from JustaSqu1d/patch-1

chore(README): update versions availability

JustaSqu1d committed:
feat: camera movement smoothing

This makes the camera less jittery and better for user experience

Co-Authored-By: scorbett123 <50634068+scorbett123@users.noreply.github.com>

MarvionKirito committed:
Merge pull request #7 from JustaSqu1d/main

feat: camera movement smoothing

MarvionKirito committed:
Improved shielding mechanics.

MarvionKirito committed:
Updated to 1.19.4

MarvionKirito committed:
Fixed altoclef crashing if quitting the world while the task is running.

JustaSqu1d committed:
docs(README): bump supported versions from 1.19.3 to 1.19.4

MarvionKirito committed:
Merge branch '1.19.4--marvion' into main

MarvionKirito committed:
Merge pull request #8 from JustaSqu1d/patch-1

docs(README): bump supported versions from 1.19.3 to 1.19.4

JamesGreen31 reviewed (approved):
Couldn’t all this be accomplished by just checking if Altoclef is in game and or is not at a loading screen? I think that this can be simplified on the outside, but I don’t see any issues.

JamesGreen31 reviewed (changes requested):
Couldn’t this all be accomplished by checking if altoclef is in game and or at a loading screen? I think this can be optimized better, but nothings wrong with this.


Fix the gradle workflow and you shouldn’t have any issues

MarvionKirito committed:
Fixed Altoclef from crashing by checking if the player is in-game before quitting.

JamesGreen31 commented on src/main/java/adris/altoclef/chains/FoodChain.java:
```diff
@@ -86,19 +86,17 @@ public float getPriority(AltoClef mod) {
             stopEat(mod);
             return Float.NEGATIVE_INFINITY;
         }
+        if (!AltoClef.inGame()) {
```
This may need to be in onTick(). See UserTaskChain

JamesGreen31 replied:
Also check the workflow--gradle isn't building

JamesGreen31 reviewed (changes requested):
Unable to find baritone -- this will break your branch if merged as gradle can't build the artifacts. See the workflow for more details.

MarvionKirito committed:
Removed AltoClef.inGame() in FoodChain and fixed gradle workflow

MarvionKirito committed:
Fixed altoclef crashing if quitting the world while the task is running.

JamesGreen31 commented on build.gradle:
```diff
@@ -43,14 +43,14 @@ dependencies {
-    //    modImplementation "cabaletta:baritone-unoptimized-fabric:1.18.2"
-    //    include "cabaletta:baritone-unoptimized-fabric:1.18.2"
-    //}
+    if (getProject().hasProperty("altoclef.development")) {
+        // Must run build from baritone-plus once
+        modImplementation 'baritone-api-fabric:baritone-unoptimized-fabric-1.19.4-beta1'
+        include "baritone-api-fabric:baritone-unoptimized-fabric-1.19.4-beta1"
+    } else {
+        modImplementation "cabaletta:baritone-unoptimized-fabric:1.19.4-beta1"
+        include "cabaletta:baritone-unoptimized-fabric:1.19.4-beta1"
+    }
```
I'm not sure you can just grab Cabaletta like this. Alto uses its own baritone and as such, needs the unoptimized jar. That's probably why the gradle is failing on the workflow.

JamesGreen31 replied:
More specifically, it won't be able to actually use the cabaletta baritone because our baritone actually references altoclef

MarvionKirito replied:
It's for my maven.

JamesGreen31 commented on build.gradle:
```diff
@@ -14,8 +14,8 @@ group = project.maven_group
     maven {
-        name = "gaucho-matrero-repo"
-        url = "https://gaucho-matrero.github.io/maven/"
+        name = "MarvionKiritoRepo"
+        url = "https://github.com/MarvionKirito/maven"
```
Let's keep the maven repo on gauchomatrero. If you were to ever leave the project, we would be ******

JamesGreen31 replied:
Once we get this working, I will merge this and it will become main

JamesGreen31 commented on gradle.properties:
```diff
@@ -12,7 +12,7 @@ minecraft_version=1.19.4
 loader_version=0.14.18
 # Mod Properties
-mod_version=1.19.4-beta1
+mod_version=1.19.4-beta2
```
You version the mod as beta2 but in the baritone settings you look for beta1.

JamesGreen31 reviewed (changes requested):
If gradle doesn't work then the whole thing won't make artifacts and will break your branch.

I'll have taco take a look to see if we can find a way to get this fixed.

dolphin-cat committed:
fix butler not working

dolphin-cat committed:
Addition of windows command line build/run instructions

dolphin-cat committed:
Update develop.md

Co-authored-by: JustaSqu1d <overenchanted.gaming@gmail.com>

dolphin-cat committed:
Update src/main/java/adris/altoclef/eventbus/events/ChatMessageEvent.java

Co-authored-by: JustaSqu1d <overenchanted.gaming@gmail.com>

dolphin-cat committed:
Update src/main/java/adris/altoclef/butler/Butler.java

Co-authored-by: JustaSqu1d <overenchanted.gaming@gmail.com>

dolphin-cat committed:
JustaSqu1d's typo

Co-authored-by: JustaSqu1d <overenchanted.gaming@gmail.com>

MarvionKirito committed:
Merge pull request #9 from dolphin-cat/main

fix butler not working

MarvionKirito committed:
Updated maven link

dolphin-cat committed:
(run command on death)[https://discord.com/channels/846150381213843497/1092783764117131356/1092783771629125652]

dolphin-cat committed:
Merge pull request #2 from MarvionKirito/main

.

dolphin-cat committed:
(run command on death)[https://discord.com/channels/846150381213843497/1092783764117131356/1092783771629125652] part 2

dolphin-cat committed:
add support for {deathmessage} placeholder

dolphin-cat committed:
Allow multiple commands

dolphin-cat committed:
Update Settings.java

dolphin-cat committed:
Delete PlayerDeathEvent.java

dolphin-cat committed:
Update DeathMenuChain.java

MarvionKirito committed:
Merge pull request #10 from dolphin-cat/main

run command on respawn

MarvionKirito committed:
Updated maven link

MarvionKirito committed:
Versions have been updated

MarvionKirito committed:
Merge remote-tracking branch 'origin/main'

MarvionKirito committed:
Fixed flinting portal error and reduced the occurrence of accidental falls

MarvionKirito committed:
Fixed flinting portal error
Reduced the occurrence of accidental falls

MarvionKirito committed:
Merge remote-tracking branch 'origin/main'

MarvionKirito committed:
Implementing self care task

MarvionKirito committed:
Implementing self care task pt. 2

MarvionKirito committed:
Implementing self-care task pt. 3

MarvionKirito committed:
Implementing self-care task pt. 4

MarvionKirito committed:
Continue diamond tool set

MarvionKirito committed:
Implementing self-care task pt. 5

MarvionKirito committed:
Implement self-care task

MarvionKirito committed:
Implement self-care task

MarvionKirito committed:
Fixing item getting stuck at cursor when picking up the item from container

MarvionKirito committed:
Fixed item getting stuck at cursor when picking up the item from container

MarvionKirito committed:
Clean up code

MarvionKirito committed:
Fixed sometimes there's a warning showing in chat saying "BlockTracker: Not tracking block right now."

Aldias7 committed:
Update MarvionBeatMinecraftTask.java

more speed

MarvionKirito committed:
Updated to 1.20.1 for bug fixes and improvements

MarvionKirito committed:
Updated to 1.20.1 for bug fixes and improvements

MarvionKirito committed:
Merge pull request #13 from Aldias7/patch-3

Update MarvionBeatMinecraftTask.java

MarvionKirito committed:
Fixed killing baby mobs while obtaining food

MarvionKirito committed:
Fixed killing baby mobs while obtaining food

MarvionKirito committed:
Merge remote-tracking branch 'MarvionAltoclefPrivate/main'

MarvionKirito committed:
Merge remote-tracking branch 'MarvionAltoclefPrivate/main'

MarvionKirito committed:
Fixed running away from hostile loop

MarvionKirito committed:
Improved CollectBlazeRodsTask

MarvionKirito committed:
Improved defense system

MarvionKirito committed:
Fixed the issue where the bot wouldn't reach its position after being knocked back while placing the bed to defeat the Ender Dragon

MarvionKirito committed:
Improved accuracy

MarvionKirito committed:
Added a feature where the bot will now attempt to eliminate all aggressive Endermen in the End

MarvionKirito committed:
Fixed the issue of crashing when the bot dies while any task is running

MarvionKirito committed:
Resolved the action of killing baby pigs when obtaining pork chops in the MarvionBeatMinecraftTask

MarvionKirito committed:
I have added a feature where the bot will wait for chunks to load at the end before taking any actions

MarvionKirito committed:
Updated to beta 2

JustaSqu1d committed:
feat: better enderman-hunting logic

JustaSqu1d committed:
feat: use better hunt logic

JustaSqu1d committed:
perf: don't hit an extra time

JustaSqu1d committed:
feat: use golden helmet instead of golden boots

Golden helmet requires one more ingot to craft, but provides and extra armor bar.

JustaSqu1d committed:
feat: add vindicators to dangerous mobs

JustaSqu1d committed:
feat: collect obsidian in nether by trading

Solves an edge case where the original portal is destroyed

JustaSqu1d committed:
feat: don't pick up eyes if there is enough

JustaSqu1d committed:
refactor: use constants for clarity

MarvionKirito committed:
Merge pull request #14 from JustaSqu1d/feat/enderman-hunt

feat: enderman hunt

MarvionKirito committed:
Merge pull request #15 from JustaSqu1d/feat/various-slight-optimizations

feat: various slight optimizations

MarvionKirito committed:
Fixed commanding butler

Jacoblightning committed:
quick and dirty fix to netherite

This is some of my first java experience. I added a quick way to fix the new netherite upgrade templates.

Jacoblightning committed:
IT WORKS??!?!?

I finally got a working version.

MarvionKirito committed:
Merge pull request #16

Smithing Fix

MarvionKirito committed:
Updated to 1.20.2

MarvionKirito committed:
Optimized entity interact.

MarvionKirito committed:
Optimized interacting with block.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
Fixed obtaining recipe and a little refactoring.

MarvionKirito committed:
Fixed obtaining recipe and a little refactoring.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
Fixed obtaining recipe and a little refactoring.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
Optimized building nether portal frame.

MarvionKirito committed:
A little refactoring.

MarvionKirito committed:
Clean up.

MarvionKirito committed:
Version correction.

MarvionKirito committed:
Merge remote-tracking branch 'MarvionAltoclefPrivate/main'

JustaSqu1d committed:
chore(gitignore): add visual studio files

JustaSqu1d committed:
feat: add smooth look

If block-placing/breaking isn't needed, `withBaritone` can be disabled for better viewing experience.

JustaSqu1d committed:
feat: optimize end fight

- move farther away from fountain to create higher perching changes

JustaSqu1d committed:
feat: "shoot arrow to an entity" task

JustaSqu1d committed:
fix: lapis requirement iron -> stone

The requirement for breaking lapis is actually stone, not iron [1].

1. https://minecraft.fandom.com/wiki/Lapis_Lazuli_Ore#Breaking

JustaSqu1d committed:
feat: add `@test arrow`

JustaSqu1d committed:
feat: ignore projectiles that will not hit player

JustaSqu1d committed:
chore: standarize command description style

JustaSqu1d committed:
chore: simplify code

MarvionKirito committed:
Look up when the dragon is dying.

MarvionKirito committed:
Just removed some codes.

MarvionKirito committed:
Kill angry endermen first.

MarvionKirito committed:
1.20.4 update

MarvionKirito committed:
1.20.5 update

MarvionKirito committed:
1.20.6 update

MarvionKirito committed:
1.21 update
